src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -433,6 +532,342 @@ def can_create_clusterrolebindings(configuration):
+    return f'\033[1m{colorama.Style.BRIGHT}{colorama.Fore.BLUE}{the_link}{colorama.Style.RESET_ALL}'
+
+
+def format_bright(msg):
+    return f'\033[1m{colorama.Style.BRIGHT}{msg}{colorama.Style.RESET_ALL}'
+
+
+def display_diagnostics_report(kubectl_prior):   # pylint: disable=too-many-statements
+    if not which('kubectl'):
+        raise ValidationError('Can not find kubectl executable in PATH')
+    subprocess_cmd = kubectl_prior + ["get", "node", "--no-headers"]
+    nodes = subprocess.check_output(
+        subprocess_cmd,
+        universal_newlines=True)
+    logger.debug(nodes)
+    node_lines = nodes.splitlines()
+    ready_nodes = {}
+    for node_line in node_lines:
+        columns = node_line.split()
+        logger.debug(node_line)
+        if columns[1] != "Ready":
+            logger.warning("Node %s is not Ready. Current state is: %s.", columns[0], columns[1])
+        else:
+            ready_nodes[columns[0]] = False
+
+    logger.debug('There are %s ready nodes in the cluster', str(len(ready_nodes)))
+
+    if not ready_nodes:
+        logger.warning('No nodes are ready in the current cluster. Diagnostics info might not be available.')
+
+    network_config_array = []
+    network_status_array = []
+    apds_created = False
+
+    max_retry = 10
+    for retry in range(0, max_retry):
+        if not apds_created:
+            subprocess_cmd = kubectl_prior + ["get", "apd", "-n", "aks-periscope", "--no-headers"]
+            try:
+                apd = subprocess.check_output(subprocess_cmd, universal_newlines=True)
+            except subprocess.CalledProcessError as ex:
+                logger.debug(f"Exception while running {subprocess_cmd}: {ex.returncode}, {ex.output}. Retrying...")
+                continue
+
+            apd_lines = apd.splitlines()
+            if apd_lines and 'No resources found' in apd_lines[0]:
+                apd_lines.pop(0)
+
+            print("Got {} diagnostic results for {} ready nodes{}\r".format(len(apd_lines),
+                                                                            len(ready_nodes),
+                                                                            '.' * retry), end='')
+            if len(apd_lines) < len(ready_nodes):
+                time.sleep(3)
+            else:
+                apds_created = True
+                print()
+        else:
+            for node_name in ready_nodes:
+                if ready_nodes[node_name]:
+                    continue
+                apdName = "aks-periscope-diagnostic-" + node_name
+                try:
+                    subprocess_cmd = kubectl_prior + ["get", "apd", apdName, "-n", "aks-periscope", "-o=jsonpath={.spec.networkconfig}"]
+                    network_config = subprocess.check_output(
+                        subprocess_cmd,
+                        universal_newlines=True)
+                    logger.debug('Dns status for node %s is %s', node_name, network_config)
```

Comment:
@sophsoph321 should we also add error in red when we find out that DNS is not working on user's cluster?